httprange/httprange.go

```diff
@@ -0,0 +1,159 @@
+// Package httprange provides RFC7233 range parsing.
+package httprange
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+)
+
+// Bytes is the bytes range type specified in the RFC.
+type Bytes struct {
+	Start     int64
+	End       int64
+	Length    int64
+	Satisfied bool
+}
+
+func (b Bytes) fmtRequest() (string, error) {
+	switch {
+	case b.Start < 0 && b.End == -1:
+		return fmt.Sprintf("%d", b.Start), nil
```

Comment:
Here, and later parts of the function would probably be faster using `strconv.Itoa`, since format-string parsing is pretty slow. Is it worth checking this? Not sure if the required string concat later (e.g. `strconv.Itoa(b.Start) + "-"`) would be slower or not.